### ISSUE TITLE
Can't run any cs or csx file - it wants to load the .NET 7.0.4 Microsoft.CSharp.dll specifically

### ISSUE THREAD

**Issue by xparadoxical** [roslynpad/roslynpad#467]
**Describe the bug**
I have .NET SDK 7.0.203, which corresponds to .NET 7.0.5. `C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\7.0.5\ref\net7.0\Microsoft.CSharp.dll` exists, but RoslynPad seems to want the dll for .NET 7.0.4 specifically, so it won't let me run anything or even see errors. The settings in the bar at the top (version and "Set platform as default") are not respected.
```
Restore Error
System.IO.FileNotFoundException: Could not find a part of the path 'C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\7.0.4\ref\net7.0\Microsoft.CSharp.dll'.
File name: 'C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\7.0.4\ref\net7.0\Microsoft.CSharp.dll'
 ---> System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\7.0.4\ref\net7.0\Microsoft.CSharp.dll'.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Roslyn.Utilities.StandardFileSystem.OpenFile(String filePath, FileMode mode, FileAccess access, FileShare share)
   at Roslyn.Utilities.CommonCompilerFileSystemExtensions.OpenFileWithNormalizedException(ICommonCompilerFileSystem fileSystem, String filePath, FileMode fileMode, FileAccess fileAccess, FileShare fileShare)
   at Roslyn.Utilities.CommonCompilerFileSystemExtensions.OpenFileWithNormalizedException(ICommonCompilerFileSystem fileSystem, String filePath, FileMode fileMode, FileAccess fileAccess, FileShare fileShare)
   at Microsoft.CodeAnalysis.MetadataReference.CreateFromFile(String path, MetadataReferenceProperties properties, DocumentationProvider documentation)
   at RoslynPad.Roslyn.RoslynHost.CreateMetadataReference(String location) in C:\src\github\aelij\RoslynPad\src\RoslynPad.Roslyn\RoslynHost.cs:line 109
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.ToArray()
   at System.Collections.Immutable.ImmutableArray.CreateRange[T](IEnumerable`1 items)
   at RoslynPad.Build.ExecutionHost.<>c__DisplayClass106_0.<<RestoreAsync>g__ReadReferencesAsync|2>d.MoveNext() in C:\src\github\aelij\RoslynPad\src\RoslynPad.Build\ExecutionHost.cs:line 742
--- End of stack trace from previous location ---
   at RoslynPad.Build.ExecutionHost.<>c__DisplayClass106_0.<<RestoreAsync>g__DoRestoreAsync|0>d.MoveNext() in C:\src\github\aelij\RoslynPad\src\RoslynPad.Build\ExecutionHost.cs:line 710
```

**To Reproduce**
I'm guessing:
- don't have .NET SDK 7.0.202 (.NET 7.0.4) installed
- open any file in RoslynPad

**Expected behavior**
It uses the selected SDK, v7.0.203 with .NET 7.0.5.

**Version**
- OS: win11 22H2 22621.1555
- Build (or commit ID if built from source): v17.0.0.0 from MS Store, updated on 2022-12-08
```
>dotnet --list-sdks
7.0.203 [C:\Program Files\dotnet\sdk]

>dotnet --list-runtimes
Microsoft.AspNetCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Comment by xparadoxical** [comment]
This is the first time I checked if RoslynPad works since April 27th, the day I created this issue on, and it does work now.

On April 21st, I updated Visual Studio. Maybe Microsoft.NETCore.App.Ref\7.0.4 got updated to 7.0.5 (based on March 13th being the modification date of some of my VS files and on [the release date of 7.0.4](https://dotnet.microsoft.com/en-us/download/dotnet/7.0)) and RoslynPad somehow was not picking this change up, even though I've done several system restarts during the 6 days that have passed? Probably something here can be improved, so I'm not closing this.
